src/sourceItemWrapper.js

```diff
@@ -663,32 +663,33 @@ class SourceItemWrapper extends ItemWrapper {
                 'loading',
                 Wikicite.getString('wikicite.source-item.add-identifier.progress.loading')
             );
-
             try {
                 if (identifiers.length > 0) {
-                    let citations = [];
                     await Zotero.Schema.schemaUpdatePromise;
-                    for (const identifier of identifiers) {
-                        let translation = new Zotero.Translate.Search();
+                    let citations = await Promise.all(identifiers.map(async (identifier) => {
```

Comment:
Maybe add a comment explaining that we are "getting items asynchronously in parallel" for future reference.